examples/point-lighting.html

```diff
@@ -0,0 +1,187 @@
+                // This vector indecates the "brightest point"; A "refrence vector" if you will.
+                vec3 halfVector = normalize(surfaceToLightDirection + surfaceToViewDirection);
+
+                // Then we can get the brightness at any point by seeing "how similar" 
+                // the surface normal is to the refrence vector. 
+                float light = dot(normal, surfaceToLightDirection);
+                
+                // By raising the specular vector to a power we can control the intensity
+                // of the light
+                float specular = 0.0;
+                if (light > 0.0) {
+                    specular = pow(dot(normal, halfVector), u_shininess * 100.0);
+                }
+
+                // Mapping textures
+                vec3 colMap = texture2D(colMap, v_uv).rgb;
+                vec3 spec = texture2D(specMap, v_uv).rgb;
+                gl_FragColor.rgb = colMap;
+
+                // Add Point Lighting
+                gl_FragColor.rgb *= light;
+                // Add Specular Highlights
+                gl_FragColor.rgb += specular * spec;
+            }
+            `;
+
+            {
+                const renderer = new Renderer();
+                const gl = renderer.gl;
+                document.body.appendChild(gl.canvas);
+                gl.clearColor(0.05, 0.05, 0.05, 1);
+
+                const camera = new Camera(gl);
+                camera.position.set(0, 0, 2);
+
+                const controls = new Orbit(camera, {
+                    target: new Vec3(0, 0, 0),
+                });
+
+                function resize() {
+                    renderer.setSize(window.innerWidth, window.innerHeight);
+                    camera.perspective({
+                        aspect: gl.canvas.width / gl.canvas.height,
+                    });
+                }
+                window.addEventListener('resize', resize, false);
+                resize();
+
+                const scene = new Transform();
+                const geometry = new Sphere(gl, { widthSegments: 64 });
+
+                // Make A point Light
+                const light = new Vec3(20, 30, 60);
+
+                // Color Map
+                const colMap = new Texture(gl);
+                const img = new Image();
+                img.onload = () => (colMap.image = img);
+                img.src = 'assets/earth.jpg';
+
+                // Specular Map
+                const specMap = new Texture(gl);
+                const specMap_img = new Image();
+                specMap_img.onload = () => (specMap.image = specMap_img);
+                specMap_img.src = 'assets/earth_specular.jpg';
+
+                const program = new Program(gl, {
+                    vertex: VERTEX_SHADER,
+                    fragment: FRAGMENT_SHADER,
+                    uniforms: {
+                        u_lightWorldPosition: { value: light }, // Position of the Light
+                        u_shininess: { value: 1.5 },
```

Comment:
in future could add light color and maybe radisu to control how far it goes

Reply:
Yes that could be added for sure, but since this is an example, i tried to keep it minimal and simple.